src/validation.cpp

```diff
@@ -3160,6 +3160,24 @@ bool ProcessNewBlockHeaders(const std::vector<CBlockHeader>& headers, CValidatio
     return true;
 }
 
+static CDiskBlockPos WriteBlockToDisk(const CBlock& block, int nHeight, const CChainParams& chainparams, const CDiskBlockPos* dbp) {
+    unsigned int nBlockSize = ::GetSerializeSize(block, SER_DISK, CLIENT_VERSION);
+    CDiskBlockPos blockPos;
+    if (dbp != NULL)
+        blockPos = *dbp;
+    if (!FindBlockPos(blockPos, nBlockSize+8, nHeight, block.GetBlockTime(), dbp != NULL)) {
+        error("AcceptBlock(): FindBlockPos failed");
+        return CDiskBlockPos();
+    }
+    if (dbp == NULL) {
+        if (!WriteBlockToDisk(block, blockPos, chainparams.MessageStart())) {
```

Comment:
Strange or not, it does care about ut. It seems more strange for this new function to take the whole CChainParams class only for one getter, no?